src/api/firebase.js

```diff
@@ -17,7 +17,9 @@ export function useShoppingListData(listId) {
 	const [data, setData] = useState(initialState);
 
 	useEffect(() => {
-		if (!listId) return;
+		// "/" is used by Firestore as a path separator and causes the app to crash when enter into the token field
+		// this restriction prevents communication with the database when "/" is used
+		if (!listId || listId.includes("/")) return;
 
```

Comment:
Good catch with this!

Reply:
We think we might have now a more optimal solution.
Right now, the validation process impacts the UI (disables the button and triggers error messages) but it does not prevent `useShoppingListData` from contacting Firestore to check whether the token exists as a collection.
The function returns immediately only if the listId is falsy or the “/” is used:
`if (!listId || listId.includes("/")) return;`
We could turn any invalid input into a falsy value by conditionally supplying the token to the function inside the Home component, like this:
`const sharedListData = useShoppingListData(isTokenValid ? token : null);`
(This also eliminates the need for the `listId.includes("/")` condition).

Reply:
Definitely seems more optimal to check for valid data before making the API call. Good job!